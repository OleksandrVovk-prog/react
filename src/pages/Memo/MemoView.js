import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import MemoComponent from './components/MemoComponent';
import Button from '../../components/Button/Button';
import styles from './sass/Memo.module.scss';

export default function MemoView({
  title, production, poster, handleClick, clickCounter, changeTitle
}) {
  const { t } = useTranslation();
  return (
    <div className={styles.memo}>
      <MemoComponent
        title={title}
        production={production}
        poster={poster}
      />
      <div className={styles.events}>
        <p className={styles.update}>
          {t('likes')}
          &nbsp;
          {clickCounter}
        </p>
        <Button
          title={t('like')}
          buttonClass={`${styles.update} ${styles.updateBtn}`}
          handleClick={handleClick}
          id="click-handler-button"
        />
        <input
          type="text"
          className={styles.update}
          value={title}
          onChange={changeTitle}
          data-testid="change-title-input"
        />
      </div>
    </div>
  );
}

MemoView.propTypes = {
  title: PropTypes.string.isRequired,
  production: PropTypes.string.isRequired,
  poster: PropTypes.string.isRequired,
  clickCounter: PropTypes.number.isRequired,
  handleClick: PropTypes.func.isRequired,
  changeTitle: PropTypes.func.isRequired,
};
