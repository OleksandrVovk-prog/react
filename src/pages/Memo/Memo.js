import { useState } from 'react';
import MemoView from './MemoView';

export default function Memo() {
  const [clickCounter, setClickCounter] = useState(0);
  const [title, setTitle] = useState('John Carter');
  const [production] = useState('Walt Disney Pictures');
  const [poster] = useState('https://m.media-amazon.com/images/M/MV5BMDEwZmIzNjYtNjUwNS00MzgzLWJiO'
  + 'GYtZWMxZGQ5NDcxZjUwXkEyXkFqcGdeQXVyNTIzOTk5ODM@._V1_SX300.jpg');

  const handleClick = () => {
    setClickCounter((count) => count + 1);
  };

  const changeTitle = (event) => {
    setTitle(event.target.value);
  };

  return (
    <MemoView
      title={title}
      production={production}
      poster={poster}
      clickCounter={clickCounter}
      handleClick={handleClick}
      changeTitle={changeTitle}
    />
  );
}
