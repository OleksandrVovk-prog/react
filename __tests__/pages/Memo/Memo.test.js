import React from 'react';
import userEvent from '@testing-library/user-event';
import { render } from '../../settings/test-utils';
import Memo from '../../../src/pages/Memo/Memo';

describe('<Memo />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = render(<Memo />);
  });
  test('should render Memo component', () => {
    const { asFragment } = wrapper;
    expect(asFragment(wrapper)).toMatchSnapshot();
  });

  test('should change input value', async () => {
    const { getByTestId } = wrapper;
    const Input = getByTestId(/change-title-input/i);
    await userEvent.type(Input, 'mock value');
    expect(Input.value).toEqual('John Cartermock value');
  });

  test('should increase initial count value', async () => {
    const { getByTestId, getByText } = wrapper;
    const Button = getByTestId(/click-handler-button/i);
    expect(getByText(/0/i)).toBeInTheDocument();
    await userEvent.click(Button);
    expect(getByText(/1/i)).toBeInTheDocument();
  });
});
