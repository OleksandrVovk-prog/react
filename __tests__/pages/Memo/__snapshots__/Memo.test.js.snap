// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Memo /> should render Memo component 1`] = `
<DocumentFragment>
  <div
    class="memo"
  >
    <div
      class="info"
    >
      <p
        class="appIntro"
      >
        Title: John Carter
      </p>
      <p
        class="appIntro"
      >
        Production: Walt Disney Pictures
      </p>
      <img
        alt="John Carter"
        class="appBanner"
        src="https://m.media-amazon.com/images/M/MV5BMDEwZmIzNjYtNjUwNS00MzgzLWJiOGYtZWMxZGQ5NDcxZjUwXkEyXkFqcGdeQXVyNTIzOTk5ODM@._V1_SX300.jpg"
      />
    </div>
    <div
      class="events"
    >
      <p
        class="update"
      >
        Likes: 0
      </p>
      <button
        class="update updateBtn"
        data-testid="click-handler-button"
        type="button"
      >
        Like
      </button>
      <input
        class="update"
        data-testid="change-title-input"
        type="text"
        value="John Carter"
      />
    </div>
  </div>
</DocumentFragment>
`;
